linode/ipv6range/framework_resource.go

```diff
@@ -0,0 +1,222 @@
+package ipv6range
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_ipv6_range",
+				IDType: types.StringType,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOpts := linodego.IPv6RangeCreateOptions{
+		PrefixLength: int(data.PrefixLength.ValueInt64()),
+	}
+
+	if !data.LinodeId.IsNull() && !data.LinodeId.IsUnknown() {
+		createOpts.LinodeID = int(data.LinodeId.ValueInt64())
+	} else if !data.RouteTarget.IsNull() && !data.RouteTarget.IsUnknown() {
+		createOpts.RouteTarget = strings.Split(data.RouteTarget.ValueString(), "/")[0]
+	} else {
+		resp.Diagnostics.AddError(
+			"Failed to create ipv6 range.",
+			"Either linode_id or route_target must be specified.",
+		)
+		return
+	}
+
+	ipv6range, err := client.CreateIPv6Range(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to create ipv6 range.",
+			err.Error(),
+		)
+		return
+	}
+
+	data.ID = types.StringValue(strings.TrimSuffix(
+		ipv6range.Range,
+		fmt.Sprintf("/%d", createOpts.PrefixLength)))
+
+	// We make the GetIPv6Range API call here because the CreateIPv6Range API endpoint
+	// only returns two fields for the newly created range (range and route_target).
+	// We need to make a second call out to the GET endpoint to populate more
+	// computed fields (region, is_bgp, linodes).
+	ipv6rangeR, diag := getIPv6Range(ctx, client, data.ID.ValueString())
+
+	if diag != nil {
+		if diag.Summary() == "IPv6 range does not exist." {
+			resp.State.RemoveResource(ctx)
+		}
+		resp.Diagnostics.Append(diag)
+		return
+	}
+
+	resp.Diagnostics.Append(data.parseIPv6RangeResourceDataComputedAttrs(ctx, ipv6rangeR)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Read(
+	ctx context.Context,
+	req resource.ReadRequest,
+	resp *resource.ReadResponse,
+) {
+	var data ResourceModel
+	client := r.Meta.Client
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ipv6range, diag := getIPv6Range(ctx, client, data.ID.ValueString())
+
+	if diag != nil {
+		if diag.Summary() == "IPv6 range does not exist." {
```

Comment:
It might be better for this error string to be a constant so we can be confident that the errors match

Reply:
```suggestion
		if diag.Summary() == ipv6rangeNotExist {
```